libnavigation-core/src/main/java/com/mapbox/navigation/core/adasis/Segment.kt

```diff
@@ -0,0 +1,49 @@
+package com.mapbox.navigation.core.adasis
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+
+/**
+ * Segment message options
+ *
+ * @param options common options for segment
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+class Segment(
```

Comment:
internal constructor + builder
Otherwise it'll be more difficult to extend the API.
But probably also not critical for the snapshot.